pkg/storegateway/indexheader/index/postings.go

```diff
@@ -404,10 +434,14 @@ func (t *PostingOffsetTableV2) LabelValues(name string, filter func(string) bool
 			d.Skip(skip)
 		}
 		s := yoloString(d.UnsafeUvarintBytes()) // Label value.
-		if filter == nil || filter(s) {
-			// Clone the yolo string since its bytes will be invalidated as soon as
-			// any other reads against the decoding buffer are performed.
-			values = append(values, strings.Clone(s))
+		if strings.HasPrefix(s, prefix) {
+			if filter == nil || filter(s) {
+				values = append(values, strings.Clone(s))
```

Comment:
nit: We lost the comment about why `strings.Clone(s)` is used. Was that on purpose?

Reply:
It wasn't on purpose, but the good thing is that I actually added it back because a test failed, and `test > comment` :D

I'll add it back.

Reply:
Added back in https://github.com/grafana/mimir/pull/4055/commits/18ecccc169aec596abacd01175e015507f17e5e0